### ISSUE TITLE
Improved documentation about using GPX notes feature.

### ISSUE THREAD

**Issue by meiphoo7-Mae** [Helium314/StreetComplete#679]
**Use case**

In the area where I tag there are a lot of projects that will a take a long time before they're completed. So I wanted to try the GPX notes feature to store notes for my personal use (instead of notes that end up publicly on OSM). In the app this is called 'Allow creating custom quests' and it says amongst other things 'Note that you'll need to enable custom quests'.
So I followed that suggestion and created a custom quest.

Now there are several problems:
1)  If I create a note, no matter if it is GPX or OSM, as soon as it is saved the note disappears immediately. Under Settings > Quests, I found the option 'Show all notes' which I've activated. When I select the option Settings > Notes > 'Get GPX notes and attached photos and tracks', I can see that my freshly created GPX note is in the file so the note is indeed present. I know that OSM hides notes for the user that created the note but I assume SCEE is not prone to that.
2) When creating a custom quest, as mentioned above, something has to be put in the field 'Filtering details'. How is one supposed to know what should be put here, given the fact that any documentation is completely non-existent both here and on OSM wiki? Pressing the white 'i' in the blue box opens a text about element selection but still it remains unclear what the name of the GPX-note element is.
3) Typing things in the field mentioned above, makes me almost want to throw my phone against the wall because literally after each character that is typed or deleted an annoying message pops up with something like 'ERROR' that covers up the keyboard. Is it really necessary to show that dozens of times?
And even when the app is killed the message remains popping up until it is apparently done with things.

**Proposed Solution**

Can certain features be documented in a manner that is understandable for a general public?
A solution for remarks 1) and 2) above also works for me.

**Comment by cpelley** [comment]
I also can't figure this out.  I have tried a few versions in case it was a bug.  Creating a gpx note just doesn't seem to show anything on the map no matter what options I play with.

Cheers

**Comment by mnalis** [comment]
> - If I create a note, no matter if it is GPX or OSM, as soon as it is saved the note disappears immediately. Under Settings > Quests, I found the option 'Show all notes' which I've activated. [...]  I know that OSM hides notes for the user that created the note but I assume SCEE is not prone to that.

Yeah, that behaviour is AFAICT the same in SC and SCEE, and is by design (at least for OSM notes) - it hides your notes. See https://github.com/Helium314/SCEE/issues/552. I should try to provide an option to ***really*** show all OSM notes in SCEE one of those days, as that is my pet peeve too.

Workarounds are:
- logout from SCEE (if the SCEE doesn't know how you _are_, it can't hide _your_ notes)
- Once upon a time adding text `#surveyme` to the note text worked too, but that seems not to work currently (https://github.com/streetcomplete/StreetComplete/issues/6052)

>  Creating a gpx note just doesn't seem to show anything on the map no matter what options I play with.

GPX notes are only available in `.gpx` file when you download it via `Settings` / `Notes` / `Get GPX notes and attached photos and tracks`. They are AFAICT intended for post-processing in full-fledged desktop OSM editor (see https://github.com/streetcomplete/StreetComplete/issues/1992, https://github.com/streetcomplete/StreetComplete/discussions/2559), and not really to be used as reference points (i.e. visible in SCEE) while on the ground...

If it would be useful for you to have them be shown in SCEE, please open a new feature request (mentioning this issue) and explain your use case for it (and detail how you'd like the feature to look / work _exactly_).

> - When creating a custom quest, as mentioned above, something has to be put in the field 'Filtering details'. 

Uh @meiphoo7-Mae, I'm not sure where you are and how you got there where it asks you for `Filtering details`. Can you provide step-by-step instructions?

At least in SCEE 58.2, AFAIK, that `Allow creating custom quests` is used as follows:
- make sure `Custom quest` is enabled in `Quest selection and display order`
- (also, preferably enable `Quests` / `Create quests dynamically`)
- click on some regular quest you can't  currently answer but want to save it in custom quest
- choose `Uh...` / `Can't say...` / `Create custom quests`
- it asks `Enter message to be shown`
- enter the message which should be displayed in that custom quest
- Later, when you browse the map, that `Custom quest` will be shown as question mark in blue circle and display the message you specified. You can then use raw tag editor, or move the node, or remove it etc.

> - How is one supposed to know what should be put here, given the fact that any documentation is completely non-existent both here and on OSM wiki? Pressing the white 'i' in the blue box opens a text about element selection but still it remains unclear what the name of the GPX-note element is.

Yeah, documentation is sometimes lacking. We're all volunteers here. So please, if you've had problems and then found out how something works, do provide a better documentation -- in GitHub, select `Create new fork` (if you haven't already), edit the `README.md` and write some text to explain it better. Then choose `create PR`. Or, if that is too complex, open a new issue and write which text would explain it better.

> - literally after each character that is typed or deleted an annoying message pops up with something like 'ERROR' that covers up the keyboard. Is it really necessary to show that dozens of times?

I've created https://github.com/Helium314/SCEE/issues/701 for that issue. Yes, it might be improved. However nobody volunteered their time to do it in better way so far. 

---

Also please note that generally, each GitHub issue should be about only **one** (actionable) problem, or it is likely to be overlooked (or worse, completely stall as it is impossible to address all issues at once).

**Comment by meiphoo7-Mae** [comment]
Hmm, since it is a few months later now I can't remember how I invoked that Filtering Details field. Reading this all again it seems that certain things got mixed up. Apparently a GPX note is not the same as a custom quest.
What I want to achieve is largely what you, @mnalis, described above but then with no restriction at all so I like to be able to create a 'real' stand alone (and private) custom quest anywhere on the map and not only be able to pair it to a regular existing quest.
In the area where I map several redevelopment projects are going on that will at least take the next five years to complete, so I would like to use the custom quest to remind me about certain things that I should survey (even if there are no existing regular quests nearby). Creating a public note makes no sense and is polluting the map because apart from a few random users that occasionally come by I'm the only mapper in the area.

And concerning about opening a feature request:
This was supposed to be a feature request. It has a use case defined and also a proposed solution in the first post (but not a very helpful solution, I know...). But anyway: no hard feelings of course. At least you gave me some pointers about the custom quest (thank you for that!) and I realize this request is not really clearly defined given the GPX note vs custom quest mix-up.

And concerning the documentation:
It did know about contributing translations, but didn't know that is was possible for users to contribute to documentation. I'll see if I can do something useful.

**Comment by mnalis** [comment]
>   What I want to achieve is largely what you, @mnalis, described above but then with no restriction at all so I like to be able to create a 'real' stand alone (and private) custom quest anywhere on the map and not only be able to pair it to a regular existing quest.


That makes sense. I've created  https://github.com/Helium314/SCEE/issues/707 with one suggestion how to do that inside SCEE which I think should not be too hard to implement (note: it also contains workaround how you can **already** do it  inside SCEE if you want to try it out, but that workaround is ugly) 

Also, one can create custom quests outside SCEE without restrictions (as a `.csv` file, see help shown when enabling Custom quest), but that is even harder to do while on the go.

Another even more powerful alternative for creating quests would be https://github.com/Helium314/SCEE/issues/682 (but that is probably too much work, and probably less well suited for your problem)


> didn't know that is was possible for users to contribute to documentation. I'll see if I can do something useful.

Thanks, all help is really appreciated!

**Comment by mnalis** [comment]
>    didn't know that is was possible for users to contribute to documentation. I'll see if I can do something useful.

Also, we have just got [SCEE FAQ](https://wiki.openstreetmap.org/wiki/SCEE/FAQ) on the OpenStreetMap wiki, so solutions how to fix some SCEE-specific issue and explaining more common confusions how to use options can be added there by any OSM contributor directly. Everyone is welcomed to contribute!

**Comment by mnalis** [comment]
As the difference between GPX note and custom quest has been explained, ways to proposed documentation/FAQ updates posted, and https://github.com/Helium314/SCEE/issues/707 opened as possible alternative, this can be closed now (please do comment if you think it still have some actionable items which have not been addressed, and I'll reopen or open a new issue as needed)

**Comment by meiphoo7-Mae** [comment]
A bit late, but you asked a valid question that deserves a honest answer.
What I want to achieve is what you described in #707, so I agree with you.
Thank you.
